src/app/shared/utils/fields.utils.ts

```diff
@@ -122,4 +122,35 @@ export class FieldsUtils {
     return Object.assign({}, obj);
   }
 
+  showGrayBar(caseField: CaseField, element: HTMLElement) {
+    if (caseField.field_type.type !== 'Collection') {
+      let divSelector = element.querySelector('div');
+      let samePage = true;
+      let cyaSelector = false;
+      this.getConditionFields(caseField.show_condition).forEach(
+        field => { samePage = document.querySelector('#' + field) ? true : false; }
+      );
+      if (divSelector) {
+        let tempElement = divSelector.parentElement;
+        while (tempElement.parentElement) {
+          if (tempElement.tagName === 'FORM') {
+            if (tempElement.classList.contains('check-your-answers')) {
+              cyaSelector = true;
+            }
+          }
+          tempElement = tempElement.parentElement;
+        }
+        if (!cyaSelector && samePage) {
+          divSelector.classList.add('panel');
+        }
+      }
+    }
+  }
+
+  getConditionFields(condition: string): any[] {
+    let condFields = [];
```

Comment:
we should not be parsing a show condition in fields.utils.ts. This violates separation of concerns principle. fields.utils.ts should know nothing about how to parse a show condition and extract his fields. 
This knowledge is encapsulated in our class ShowCondition. 
This logic should be moved there and unit tests should be added to `conditional-show.model.spec.ts`